extensions/apps/extensions/src/components/pages/my-extensions/extension-element.tsx

```diff
@@ -0,0 +1,251 @@
+
+import { hasOwn, useRootComponentProps } from '@akashaorg/ui-awf-hooks';
+import { useGetAppsStreamQuery } from '@akashaorg/ui-awf-hooks/lib/generated/apollo';
+import {
+  AkashaAppApplicationType,
+  AkashaAppsStreamModerationStatus,
+  AppImageSource,
+} from '@akashaorg/typings/lib/sdk/graphql-types-new';
+import { ExtensionStatus } from '@akashaorg/typings/lib/ui';
+
+type Extension = {
+  id?: string;
+  name?: string;
+  displayName?: string;
+  logoImage?: AppImageSource;
+  description?: string;
+  applicationType?: AkashaAppApplicationType;
+};
+
+type ExtensionElement = {
+  extensionData: Extension;
+  showDivider?: boolean;
+  filter: { id?: string; title?: string; opt?: string };
+};
+
+export const ExtensionElement: React.FC<ExtensionElement> = ({
+  extensionData,
+  showDivider,
+  filter,
+}) => {
+  const { t } = useTranslation('app-extensions');
+  const sdk = React.useRef(getSDK());
+
+  const { navigateToModal } = useRootComponentProps();
+
+  const { data: appStreamReq } = useGetAppsStreamQuery({
+    variables: {
+      indexer: sdk.current.services.gql.indexingDID,
+      first: 1,
+      filters: {
+        where: {
+          applicationID: {
+            equalTo: extensionData.id,
+          },
+        },
+      },
+    },
+    skip: !extensionData.id,
+  });
+
+  const appStreamData =
+    appStreamReq?.node && hasOwn(appStreamReq.node, 'akashaAppsStreamList')
+      ? appStreamReq.node.akashaAppsStreamList
+      : null;
+
+  const getIconByAppType = (applicationType: AkashaAppApplicationType) => {
+    switch (applicationType) {
+      case AkashaAppApplicationType.App:
+        return <App />;
+      case AkashaAppApplicationType.Plugin:
+        return <Plugin />;
+      case AkashaAppApplicationType.Widget:
+        return <Widget />;
+    }
+  };
+
+  const getExtensionStatus = () => {
+    if (appStreamData?.edges?.length === 0) {
+      return ExtensionStatus.Draft;
+    }
+    if (appStreamData?.edges[0]) {
+      if (appStreamData?.edges[0].node?.status === null) {
+        return ExtensionStatus.Draft;
+      }
+      if (appStreamData?.edges[0].node?.status === AkashaAppsStreamModerationStatus.InReview) {
+        return ExtensionStatus.Pending;
+      }
+      if (appStreamData?.edges[0].node?.status === AkashaAppsStreamModerationStatus.Ok) {
+        return ExtensionStatus.Published;
+      }
+    }
+  };
+
+  const getStatusIndicatorStyle = () => {
+    switch (getExtensionStatus()) {
+      case ExtensionStatus.Draft:
+        return 'bg-grey6';
+      case ExtensionStatus.Published:
+        return 'bg-success';
+      case ExtensionStatus.Pending:
+        return 'bg-(errorLight dark:errorDark)';
+      default:
+        return 'bg-grey6';
+    }
+  };
+
+  const handleExtensionRemove = () => {
+    navigateToModal({
+      name: `remove-extension-confirmation`,
+      extensionId: extensionData.id,
+    });
+  };
+
+  const menuItems = (extensionStatus: string) => {
```

Comment:
```suggestion
  const menuItems = (extensionStatus: string) : MenuProps['items'] | [] => {
```
That instead of type assertions on each case.